### PR TITLE
Add Horizontal Charts (Bar & Scatter)

### PR DESCRIPTION
**Pull request by mellowfish**
Also fixed several issues with axis layout and other minor bugs.

### CODE REVIEW

**Comment by farneman** [comment]
Don't see anything obviously wrong and I've worked with this some to know it works correctly. :shipit:

File: lib/prawn/charts/horizontal_base.rb [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        with_color do
+          bounding_box at, width: width, height: height do
+
+            with_larger_font { draw_title } if config[:title].present?
+            draw_legend if config[:legend].present?
+
+            bounding_box(chart_at, width: chart_width, height: chart_height) do
+              draw_x_axis  if x[:display]
+              draw_y_axis  if y[:display]
+              # draw_y1_axis if y1[:display]
```
Comment (gray-sanborn-work) [inline lib/prawn/charts/horizontal_base.rb]: **Rocking:** Delete commented line.

File: lib/prawn/charts/horizontal_scatter.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            #   text = value_formatter.call(h[:value])
+            #   label_width = width_of(text)
+            #   label_x = point_x - radius - label_width / 2.0
+            #   label_y = point_y - 2 * radius
+            #   if index.even?
+            #     label_y = point_y + 2 * radius + label_height
+            #   end
+            #
+            #   text_box text, at: [label_x, label_y], height: bar_height, width: label_width
+            # end
```
Comment (gray-sanborn-work) [inline lib/prawn/charts/horizontal_scatter.rb]: **Rocking:** Keep or delete.

**Comment by gray-sanborn-work** [comment]
:shipit:

**Pull request merged**